### ISSUE TITLE
EIP1193 Support

### ISSUE THREAD

**Issue by gakonst** [gakonst/ethers-rs#119]
* https://github.com/tomusdrw/rust-web3/pull/414/files
* https://github.com/enolan/rust-web3/commit/447d5faba1e0f03f12f4856e05d25a4bdbfcbfb7

**Comment by hiram-abif** [comment]
Support and examples for EIP1193 would help a lot.

**Comment by bazyli** [comment]
Because I'm in quite a need, I've picked this issue up. Implementing support outside your repo, but thinking about making an official PR if I'm successful unless someone is already working on this.

**Comment by gakonst** [comment]
Sweet - supportive. Got a repo?

**Comment by bazyli** [comment]
I should finish my tests today. Will share it if successful :)

**Comment by bazyli** [comment]
@gakonst Here's my fork where I work over this feature. It's still not finished, but if you feel that I'm not going according to your coding standards, it would be nice to hear it straight away :) https://github.com/brightpoolone/ethers-rs/tree/features/eip1193

**Comment by gakonst** [comment]
This looks good directionally! Please remove any unrelated formatting changes. I think we'll also want a usage example & test, but the logic looks good to me.

**Comment by bazyli** [comment]
Sorry for the formatting thing.. My prettier did that for me :P Will revert. And yes - pub/sub is in the queue. As for testing - that's the tricky part. Not sure how I can make any unit tests mocking the whole wasm stack and access to `window.ethereum`. Meaning - I need some time before this code will be production level.

**Comment by gakonst** [comment]
>  Not sure how I can make any unit tests mocking the whole wasm stack and access to window.ethereum

Yeah I think would be hard. Maybe consider modifying/making this example use EIP1193 if a window.ethereum is detected, else fall back? https://github.com/gakonst/ethers-rs/blob/master/examples/ethers-wasm/src/lib.rs
